examples/C/src/leader-election/HeartbeatBully.lf

```diff
@@ -0,0 +1,203 @@
+/**
+ * This program models a redundant fault tolerant system where a primary node, if and when it fails,
+ * is replaced by one of several backup nodes. The protocol is described in this paper:
+ *
+ * Bjarne Johansson; Mats Rågberger; Alessandro V. Papadopoulos; Thomas Nolte, "Heartbeat Bully:
+ * Failure Detection and Redundancy Role Selection for Network-Centric Controller," Proc. of the
+ * 46th Annual Conference of the IEEE Industrial Electronics Society (IECON), 8-21 October 2020.
+ * https://doi.org/10.1109/IECON43393.2020.9254494
+ *
+ * The program has a bank of redundant nodes where exactly one is the primary node and the rest are
+ * backups. The primary node is always the one with the highest bank index that has not failed. The
+ * primary sends a heartbeat message once per second (by default). When the primary fails, a leader
+ * election protocol selects a new primary which then starts sending heartbeat messages. The program
+ * is set so that each primary fails after sending three heartbeat messages. When all nodes have
+ * failed, then the program exits.
+ *
+ * This example is designed to be run as a federated program with decentralized coordination.
+ * However, as of this writing, bugs in the federated code generator cause the program to fail
+ * because all federates get the same bank_index == 0. This may be related to these bugs:
+ *
+ * - https://github.com/lf-lang/lingua-franca/issues/1961
```

Comment:
Both of these issues have now been closed.

Reply:
The HeartbeatBully example now works federated!  Thanks all.

However, the NRP_LF.lf example still does not work. It fails with an STP violation.

Reply:
Could you create an issue for this remaining problem in the `lingua-franca` repo? Thanks!